packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -343,9 +419,18 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
     final double contentBottom = math.max(0.0, bottom - math.max(minInsets.bottom, bottomWidgetsHeight));
 
     if (hasChild(_ScaffoldSlot.body)) {
-      final BoxConstraints bodyConstraints = BoxConstraints(
+      double bodyMaxHeight = math.max(0.0, contentBottom - contentTop);
+
+      if (extendBody)
+        bodyMaxHeight += bottomWidgetsHeight;
```

Comment:
we need to make sure we don't add so much that `bodyMaxHeight` ends up bigger than the height of the scaffold itself (minus contentTop).

Reply:
The arithmetic is supposed to guarantee that so I've added an assert to protect against back sliding.